### ISSUE TITLE
Question about losses

### ISSUE THREAD

**Issue by willow-hu** [YehLi/xmodaler#31]
These days I'm trying to use multiple losses to train the model, and I intend to set weights to the losses, but I didn't find where to set weights. It seems that your code could only add them directly without weighting, thus I really hope you can add this tiny function into your code. 

BTW, there's a tiny mistake in DefaultTrainer class:
- xmodaler/engine/defaults.py, line 523
should be:
`self._write_metrics(losses_dict, data_time)`
instead of 'loss_dict'

**Comment by jianjieluo** [comment]
Hi, @willow-hu 

Thanks for your feedback. To set weights to the losses, the best practice is to add a weight variable in the class of loss and weight the loss in the forward function. For example, if you want to add a loss weight to `CrossEntropy` loss, you can rewrite the `from_config()` and `__init__()` methods in `xmodaler/losses/cross_entropy.py` like below

```python
@LOSSES_REGISTRY.register()
class CrossEntropy(nn.Module):
    @configurable
    def __init__(self, loss_weight):
        super(CrossEntropy, self).__init__()
        self.criterion = nn.CrossEntropyLoss(ignore_index=-1)
        self.loss_weight = loss_weight # add the `self.loss_weight` in `__init__()`

    @classmethod
    def from_config(cls, cfg):
        return {
            "loss_weight": cfg.LOSSES.CROSS_ENTROPY_WEIGHT # add an customed item `_C.LOSSES.CROSS_ENTROPY_WEIGHT` in `xmodaler/config/defaults.py`
        }

    def forward(self, outputs_dict):
            ...
            loss = self.criterion(logits, targets) * self.loss_weight # weight the loss in `forward()`
            ret.update({ 'CrossEntropy Loss(G)': loss })
            ...
            loss = self.criterion(logits, targets) * self.loss_weight # weight the loss in `forward()`
            ret.update({'CrossEntropy Loss(U)': loss})
            ...
```
after such modifications, multiple losses will be summed after re-weighting.

In other to keep the code clean, we won't add the `self.loss_weight` for every loss while you can add it freely according to your needs. What's more, I appreciate that you found the tiny mistake in `self._write_metrics()`, and I have fixed such tiny mistakes.

Best,
Jianjie

**Comment by willow-hu** [comment]
Thanks for ur reply! That's a clean modification.
